megamek/src/megamek/common/weapons/ArtilleryBayWeaponIndirectHomingHandler.java

```diff
@@ -607,6 +622,33 @@ protected int getCounterAV() {
         return CounterAV;
     }
     
+    /*
+     * Returns the heat generated by a large craft's weapons fire declarations during the round
+     * Used to determine whether point defenses can engage.
+     * @Param ae - the entity you wish to get heat data from
+     */
+    @Override
+    protected int getLargeCraftHeat(Entity e) {
```

Comment:
Splitting the getLargeCraftHeat() discussion from the other one, it looks like you have 7 identical implementations of this method and one different one for the WeaponHandler class. So, you definitely need to consolidate these implementations.

Also, I recently discovered that there's a convenience method in Entity called "hasETypeFlag" or something similar, that takes an int (which should be one of the ETYPE_ flags) so you don't have to fool around with manual bit comparisons.